cli/src/generate/prepare_grammar/intern_symbols.rs

```diff
@@ -147,6 +151,21 @@ impl<'a> Interner<'a> {
 
         None
     }
+
+    // In the case of a seq or choice rule of 1 element in a hidden rule, weird
```

Comment:
This seems to be a trick grammars use to expose single anonymous nodes (see list in comment above), which now no longer works (the anonymous node is no longer exposed so queries are broken).

It also seems to behave unexpectedly with using a `const` in a simple (non-seq or choice) rule as in [clojure](https://github.com/sogaiu/tree-sitter-clojure/blob/f4236d4da8aa92bc105d9c118746474c608e6af7/grammar.js#L338-L339).

Reply:
(Definitely should have been marked breaking...)

Reply:
For convenience, [here](https://github.com/sogaiu/tree-sitter-clojure/blob/f4236d4da8aa92bc105d9c118746474c608e6af7/grammar.js#L189-L194) is the definition of the `const` mentioned above to go along with the link to the rule in tree-sitter-clojure.

Reply:
I also think this should have been a separate PR and commit...

Reply:
@amaanq seems like there is some negative fallout from this change. Some grammars explicitly wanted the old behavior, in which single-child seq was not special cased, but just treated like other sequences.

Should we consider reverting that part of this PR?

Reply:
I've already gone through the ecosystem and opened PRs to the affected parsers (I know of).

I've yet to hear from one that _wants_ to have a rule wrapping an anonymous node (which seems to me to just complicate parsers for little gain). 

However, the check _should_ be made smarter to not trigger the false positives reported in #3455.

(And, yes, some way of testing the side-effects of such changes _before_ landing them would be very useful -- tree-sitter has evolved quite a bit beyond the original handful of parsers under this org that are pulled in as fixtures.)